test/Rx/Functional/Operator/ZipTest.php

```diff
@@ -0,0 +1,615 @@
+            onNext(150, 1),
+            onNext(230, 3),
+            onNext(260, 6),
+            onCompleted(400)
+        ]);
+
+        $result = $this->scheduler->startWithCreate(function () use ($e0, $e1, $e2) {
+            return $e0->zip([$e1, $e2], function ($r0, $r1, $r2) {
+                return [$r0, $r1, $r2];
+            });
+        });
+
+        $this->assertMessages([
+            onNext(230, [1, 2, 3]),
+            onNext(260, [4, 5, 6]),
+            onCompleted(420)
+        ],
+            $result->getMessages()
+        );
+
+        $this->assertSubscriptions([
+            subscribe(200, 420)
+        ],
+            $e0->getSubscriptions());
+
+        $this->assertSubscriptions([
+            subscribe(200, 420)
+        ],
+            $e0->getSubscriptions());
+
+        $this->assertSubscriptions([
+            subscribe(200, 420)
+        ],
+            $e0->getSubscriptions());
+    }
+
+    public function testZipNeverNever()
+    {
+        $o1 = new NeverObservable();
+        $o2 = new NeverObservable();
+
+        $results = $this->scheduler->startWithCreate(function () use ($o1, $o2) {
+            return $o1->zip([$o2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([], $results->getMessages());
+    }
+
+    public function testZipNeverEmpty()
+    {
+        $o1 = new NeverObservable();
+        $o2 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(210)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($o1, $o2) {
+            return $o1->zip([$o2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([], $results->getMessages());
+    }
+
+    public function testZipEmptyEmpty()
+    {
+        $o1 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(210)
+        ]);
+        $o2 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(210)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($o1, $o2) {
+            return $o1->zip([$o2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([onCompleted(210)], $results->getMessages());
+    }
+
+    public function testZipEmptyNonEmpty()
+    {
+        $o1 = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(210)
+        ]);
+        $o2 = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(215, 2),
+            onCompleted(220)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($o1, $o2) {
+            return $o1->zip([$o2], function ($x, $y) {
+                return $x + $y;
+            });
+        });
+
+        $this->assertMessages([onCompleted(215)], $results->getMessages());
```

Comment:
I was confused about this as well. But why wouldn't this complete at 210? The first observable has completed and there is nothing else queued up in that, so at 210, there is not going to be any more onNext notifications. Can it complete there? Is there a "reference" to know exactly what the behavior should be?